backend/image/test/upload.e2e-spec.ts

```diff
@@ -1,35 +1,44 @@
+import { HttpService } from '@nestjs/axios';
 import { HttpStatus, INestApplication } from '@nestjs/common';
-import { AppModule } from '../src/app.module';
-import { Test, TestingModule } from '@nestjs/testing';
 import { ConfigModule } from '@nestjs/config';
-import { UploadService } from '../src/upload/upload.service';
-import * as request from 'supertest';
+import { Test, TestingModule } from '@nestjs/testing';
 import * as fs from 'fs';
-import { emptyDir } from './helpers/file.helper';
-import { JwtAuthGuard } from '../src/shared/guards/jwt.guard';
-import { FakeAuthGuardFactory } from './mocks/jwt-guard.mock';
 import * as path from 'path';
+import * as request from 'supertest';
+import { AppModule } from '../src/app.module';
+import { JwtAuthGuard } from '../src/shared/guards/jwt.guard';
 import { pathBuilder } from '../src/shared/helpers/file-path.helpers';
 import { JwtStrategy } from '../src/shared/strategies/jwt.strategy';
+import { UploadService } from '../src/upload/upload.service';
+import { emptyDir } from './helpers/file.helper';
+import { HttpServiceMock } from './mocks/http-service.mock';
+import { FakeAuthGuardFactory } from './mocks/jwt-guard.mock';
+import { EmptyLogger } from './mocks/logger.mock';
 
 describe('UploadController (e2e)', () => {
   let app: INestApplication;
   let uploadService: UploadService;
+  let httpMock = new HttpServiceMock();
+
   const fakeGuard = new FakeAuthGuardFactory();
 
   beforeEach(async () => {
```

Comment:
beforeEach + afterEach creating the app sounds heavy, considered using beforeAll?
If changed, make sure to set httpMock.shouldFail = fail; goes into the beforeEach()